src/main/java/com/linkedin/kmf/apps/MirrorPipelineMonitor.java

```diff
@@ -70,6 +70,7 @@ public MirrorPipelineMonitor(Map<String, Object> props, String name) throws Exce
                                                                      String topic,
                                                                      String name) throws Exception {
     Map<String, Object> topicManagementProps = createTopicManagementServiceProps(props, topic);
+    topicManagementProps.put(CommonServiceConfig.ZOOKEEPER_CONNECT_CONFIG, "");
```

Comment:
In general assigning an empty string to a required config seems weird. After taking a further look a the code, I realize that you have this line is because you want to change the `PARTITIONS_TO_BROKER_RATIO_THRESHOLD` based on its current value in the `props` and the current number of brokers in each cluster. However, this reveals another problem with `MirrorPipelineMonitor`, i.e. what if the number of brokers change after the `MirrorPipelineMonitor` starts running? If SRE doubles the number of brokers in one cluster, `TopicManagementService` will double partition number in that cluster only which breaks `MirrorPipelineMonitor`, right? It seems that the only choice is to build this logic into `TopicManagementService` so that it is able to keep partition number equivalent across multiple clusters.